src/hotspot/share/services/heapDumper.cpp

```diff
@@ -2659,17 +2645,10 @@ int HeapDumper::dump(const char* path, outputStream* out, int compression, bool
   //          This is done by DumpMerger, which is performed outside safepoint
 
   DumpMerger merger(path, &writer, dumper.dump_seq());
-  Thread* current_thread = Thread::current();
-  if (current_thread->is_AttachListener_thread()) {
-    // perform heapdump file merge operation in the current thread prevents us
-    // from occupying the VM Thread, which in turn affects the occurrence of
-    // GC and other VM operations.
-    merger.do_merge();
-  } else {
-    // otherwise, performs it by VM thread
-    VM_HeapDumpMerge op(&merger);
-    VMThread::execute(&op);
-  }
+  // perform heapdump file merge operation in the current thread prevents us
```

Comment:
Nit: Better to start with a capital letter.

Reply:
Fixed.